tools/helper.psm1

```diff
@@ -90,21 +90,42 @@ function New-gRPCAutoGenCode
     )
 
     if ($Force -or -not (Test-Path "$RepoRoot/src/Messaging/FunctionRpc.cs") -or
-                   -not (Test-Path "$RepoRoot/src/Messaging/FunctionRpcGrpc.cs"))
+                   -not (Test-Path "$RepoRoot/src/Messaging/FunctionRpcGrpc.cs") -or
+                   -not (Test-Path "$RepoRoot/src/Messaging/NullableString.cs"))
     {
         Write-Log "Generate the CSharp code for gRPC communication from protobuf"
 
         Resolve-ProtoBufToolPath
 
         $outputDir = "$RepoRoot/src/Messaging"
         Remove-Item "$outputDir/FunctionRpc*.cs" -Force -ErrorAction SilentlyContinue
+        Remove-Item "$outputDir/NullableString.cs" -Force -ErrorAction SilentlyContinue
 
-        & $Script:protoc_Path $Script:claimsIdentityRpc_proto_file_Path `
+        # Generate shared types
+        foreach ($file in @(Get-ChildItem $Script:shared_dir_Path\*))
+        {
+            & $Script:protoc_Path $file.Fullname `
             --csharp_out $outputDir `
             --grpc_out=$outputDir `
             --plugin=protoc-gen-grpc=$Script:grpc_csharp_plugin_Path `
-            --proto_path=$Script:identity_dir_Path `
+            --proto_path=$Script:protobuf_dir_Path `
             --proto_path=$Script:google_protobuf_tools_Path
+        }
+
+        # Generate other types
+        foreach ($folder in @(Get-ChildItem $Script:protobuf_dir_Path\* | Where-Object{$_.PsIsContainer -and $_.FullName -notmatch 'shared'}))
```

Comment:
```suggestion
        foreach ($folder in @(Get-ChildItem $Script:protobuf_dir_Path\* -Exclude *shared* -Directory))
```

Does this work?

Reply:
Yes, we could also do that. I will update the code.